### PR TITLE
Igor/dhsb 462 ui

### PR DESCRIPTION
**Pull request by igormas**

### CODE REVIEW

File: src/components/screenSharing/services/screenSharingUiSrv.js [hunk 1]
```diff
@@ -53,16 +54,43 @@
...
 
             this.endScreenSharing = function () {
                 _endScreenSharing();
             };
 
+            this.showScreenSharingConfirmationPopUp = function(){
+                var translationsPromMap = {};
+                translationsPromMap.title = $translate('SCREEN_SHARING.SHARE_SCREEN_REQUEST');
+                translationsPromMap.content= $translate('SCREEN_SHARING.WANT_TO_SHARE',{
+                    name: "Student/Teacher"
```
Comment (assafshp) [inline src/components/screenSharing/services/screenSharingUiSrv.js]: what is this name?

File: src/components/screenSharing/services/screenSharingUiSrv.js [hunk 2]
```diff
@@ -53,16 +54,43 @@
...
+                translationsPromMap.cancelBtnTitle = $translate('SCREEN_SHARING.ACCEPT');
+                return $q.all(translationsPromMap).then(function(translations){
+                    var popUpInstance = PopUpSrv.warning(
+                        translations.title,
+                        translations.content,
+                        translations.acceptBtnTitle,
+                        translations.cancelBtnTitle
+                    );
+                    return popUpInstance.promise.then(function(res){
+                        return $q.reject(res);
```
Comment (assafshp) [inline src/components/screenSharing/services/screenSharingUiSrv.js]: is res is relevant in reject ?

File: src/components/screenSharing/services/screenSharingSrv.js [hunk 3]
```diff
@@ -36,14 +46,24 @@
...
+                    errMsg = 'ScreenSharingSrv: screen sharing is already active!!!';
+                    $log.debug(errMsg);
+                    return $q.reject(errMsg);
                 }
 
                 var initScreenSharingStatus = _getScreenSharingInitStatusByInitiator(initiator);
                 if (!initScreenSharingStatus) {
+                    errMsg = 'ScreenSharingSrv: initiator was not provided';
+                    $log.error(errMsg);
                     return $q.reject('ScreenSharingSrv: initiator was not provided');
```
Comment (assafshp) [inline src/components/screenSharing/services/screenSharingSrv.js]: why not return $q.reject(errMsg) ?

File: dist/screenSharing/screenSharing.js [hunk 4]
```diff
@@ -381,11 +482,10 @@
     'use strict';
 
     angular.module('znk.infra.screenSharing').service('ScreenSharingUiSrv',
-        ["$rootScope", "$timeout", "$compile", "$animate", function ($rootScope, $timeout, $compile, $animate) {
+        ["$rootScope", "$timeout", "$compile", "$animate", "PopUpSrv", "$translate", "$q", "$log", function ($rootScope, $timeout, $compile, $animate, PopUpSrv, $translate, $q, $log) {
```
Comment (assafshp) [inline dist/screenSharing/screenSharing.js]: can you change all " to '

File: src/components/screenSharing/services/screenSharingUiSrv.js (hunk 1 above)
Comment (igormas) [inline src/components/screenSharing/services/screenSharingUiSrv.js]: Temporary

File: src/components/screenSharing/services/screenSharingUiSrv.js (hunk 2 above)
Comment (igormas) [inline src/components/screenSharing/services/screenSharingUiSrv.js]: The pop up behaviour is opposite than what we need in screen sharing, so the answer is yes.

File: src/components/screenSharing/services/screenSharingSrv.js (hunk 3 above)
Comment (igormas) [inline src/components/screenSharing/services/screenSharingSrv.js]: Fixed

**Pull request merged**